UI alignment on Homepage, Catagory Landing, Entity pages

comzeradd opened the pull request:
Related issues: #370, #372, #369, #371, #373, #378 

Did some backend changes too. Notably I created two snippets, one for Heading and one for the Bottom banner. Both can be reusable among all templates. For now Homepage and Category Landing already refactored in order to utilize these snippets. I intend to do the same with the rest of the templates in separate PR.

akatsoulas commented:
I would suggest to pause any refactoring for now. We need to better understand the patterns that we are seeing in the project and decide what's the best way to do this moving forward. Personally, I am not sure if this is snippets, smaller functional units, StreamFields or a combination of them. We surely need small content blocks that can be used in any number of templates in any order. Snippets can't help us with that since it's just a monolithic content block.

akatsoulas commented on coss/home/models.py:
```diff
@@ -19,6 +20,96 @@ class Meta:
+class BottomBanner(models.Model):
+    title = models.CharField(blank=True, default='', max_length=50)
+    text = models.CharField(blank=True, default='', max_length=100)
+    cta_left_text = models.CharField(verbose_name='Text', blank=True, default='',
```
Do we need this in order to tackle the issues mentioned in the PR? If not, let's drop them. I believe our action plan should be first to finish all the templates that we have for phase 1 and to only add any missing functionality. When this work is complete we can start moving things around and refactor them.

For example, ideally we want a block of code that contains only a cta link and can be re-used in multiple places in a template. I would prefer to go down this road first and if it's not possible to achieve this, then revert to using multiple entries for the same field.

akatsoulas replied:
Moreover, it's better to tackle any issues/bugs separately from a refactor.

comzeradd replied:
The reason I did this here is because that banner is displayed at least on two different templates. Do you prefer I replicate all these fields in two different models for now?

akatsoulas replied:
Yes! We can design the refactor the coming week so that we can get as much as we can from reusable entities that can be abstract from the design.

akatsoulas commented:
I am pausing the review here, because the fixes are based on the refactor. Let's revert the later and I will do another pass when it's ready

comzeradd commented:
PR updated. I reverted all model refactoring, focused only on the frontend. I added a commit for an Image gallery on the Entity page, since it depends on some css changes made already in this PR.

akatsoulas commented:
Please drop the last commit and maybe open a new PR with this, it's not relevant to either the category landing or the homepage.

comzeradd commented:
PR updated, removing the last commit.

akatsoulas commented on coss/opensource_clubs/templates/opensource_clubs/home_page.html:
```diff
@@ -9,66 +9,52 @@
+      {% endif %}
+    </div>
+
+    {% if page.bottom_cta_title %}
```
The bottom cta and the footer have a big space in between.
![screenshot-2017-9-11 homepage](https://user-images.githubusercontent.com/900565/30272245-417da10a-96fc-11e7-91a4-67be41b935ae.png)

akatsoulas commented on coss/opensource_clubs/templates/opensource_clubs/category_landing_page.html:
```diff
@@ -1,33 +1,24 @@
-      {% for club in page.get_children %}
-        {% include 'opensource_clubs/club_card.html' with club=club.specific %}
-      {% endfor %}
+  <div class="content inverted">
```
This should be wrapped in an `if` block in case there are no children.

akatsoulas replied:
Maybe it's a good idea to have an `else` clause too, to display something. Maybe it could something simple like a text paragraph that can guide the content creators on how to add content for this page. Something along the lines of " You need to add entities too.". Thoughts?
